lib/client/api.go

```diff
@@ -4552,6 +4560,25 @@ func (tc *TeleportClient) applyAuthSettings(authSettings webclient.Authenticatio
 	if authSettings.PrivateKeyPolicy != "" && !authSettings.PrivateKeyPolicy.IsSatisfiedBy(tc.PrivateKeyPolicy) {
 		tc.PrivateKeyPolicy = authSettings.PrivateKeyPolicy
 	}
+
+	var ssoURL *url.URL
+	var err error
+	switch {
+	case authSettings.OIDC != nil:
+		ssoURL, err = url.Parse(authSettings.OIDC.IssuerURL)
+	case authSettings.SAML != nil:
+		ssoURL, err = url.Parse(authSettings.SAML.SSO)
+	case authSettings.Github != nil:
+		ssoURL, err = url.Parse(authSettings.Github.EndpointURL)
```

Comment:
Is it possible that the host coming from one of these fields is different than the host used in the SSO MFA configuration? Say the SSO login directs me to `sso.example.com` but for the MFA SSO I use `mfa-sso.example.com`.

Reply:
For SAML alone this is possible, but in practice should never happen. For SSO MFA to work as expected, the same provider (host) needs to be in use (e.g. `dev-xxxxxx.oktapreview.com`) for both login and MFA. You could technically get it working if you have two providers with the same set of users, but you'd have a lot of issues. I'll leave a comment for now in case we run into unexepcted issues in the future.